pkg/registry/core/pod/strategy.go

```diff
@@ -71,6 +71,7 @@ func (podStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 		Phase:    api.PodPending,
 		QOSClass: qos.GetPodQOS(pod),
 	}
+	pod.Spec.EphemeralContainers = nil
```

Comment:
this simply drops, instead of complaining in validation if this field is attempted to be set on create. that seems confusing ("I set this field in spec, the server accepted it, but dropped my data")... is that what we want?

Reply:
If you moved this, why is it still here?

Reply:
I added a new commit to the PR, "Updates to EphemeralContainers API", which moved it, but this conversation seems sticky to the earlier commit.

Unless you'd like to see how the PR evolved, I think it's time to squash.